src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Helpers/BackupScheduleHelpers.ps1

```diff
@@ -0,0 +1,346 @@
+		[AllowNull()]
+		[string[]]
+		${ScheduleRunDay},
+
+		[Parameter(Mandatory)]
+		[AllowEmptyString()]
+		[string]
+		${ScheduleTime},
+
+		[Parameter(Mandatory)]
+		[AllowEmptyString()]
+		[string]
+		${TimeZone}
+	)
+
+	process
+	{
+		#Write-Debug "Validating Backup Schedule Options"
+		#Write-Debug -Message $DatasourceType
+
+		$manifest = LoadManifest -DatasourceType $DatasourceType
+
+		# Validate BackupFrequency
+
+		if(-not $BackupFrequency) {
+			if ($manifest.allowedSubProtectionPolicyTypes.Count -gt 1) {
+				# SAPHANA case
+				if(-not $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunFrequency) {
+					$errormsg = "BackupFrequency cannot be empty for FullBackup"
+					throw $errormsg
+				})
+				else {
+					$BackupFrequency = $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunFrequency
+				}
+			}
+			else {
+				# AzureVM case
+
+			}
+		}
+
+		if (-not $manifest.allowedBackupFrequencies.Contains($BackupFrequency)) {
+			$allowedValues = [System.String]::Join(', ', $manifest.allowedBackupFrequencies)
+			$errormsg = "Specified BackupFrequency is not supported for DatasourceType " + $DatasourceType + "`nAllowed values are: " + $allowedValues
+			throw $errormsg
+		}
+
+		# Validate ScheduleTime
+		
+		if(-not $ScheduleTime) {
+			if ($manifest.allowedSubProtectionPolicyTypes.Count -gt 1) {
+				# SAPHANA case
+				if(-not $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunTime.Count -gt 0) {
+					$errormsg = "ScheduleTime cannot be empty for FullBackup"
+					throw $errormsg
+				}
+				else {
+					$ScheduleTime = $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunTimes[0]
+				}
+			}
+			else {
+				# AzureVM case
+			}
+		}
+
+		# Validate ScheduleRunDay
+
+		if(-not $ScheduleRunDay) {
+			if ($manifest.allowedSubProtectionPolicyTypes.Count -gt 1) {
+				# SAPHANA case
+				if(-not $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunDays) {
+					$errormsg = "ScheduleRunDay cannot be empty for FullBackup"
+					throw $errormsg
+				}
+				else {
+					$ScheduleRunDay = $Policy.SubProtectionPolicy[0].SchedulePolicy.ScheduleRunDays
+				}
+			}
+			else {
+				# AzureVM case
+			}
+		}
+
+		# Validate TimeZone
+
+		if(-not $TimeZone) {
+			if ($manifest.allowedSubProtectionPolicyTypes.Count -gt 1) {
+				# SAPHANA case
+				if(-not $Policy.Setting.TimeZone) {
+					$errormsg = "TimeZone cannot be empty for FullBackup"
+					throw $errormsg
+				}
+				else {
+					$TimeZone = $Policy.Setting.TimeZone
+				}
+			}
+			else {
+				# AzureVM case
+			}
+		}
+
+		# Validate PolicySubType
+		
+		if ($PolicySubType) {
+			if (-not $manifest.supportsEnhanced) {
+				$errormsg = "PolicySubType is not supported for DatasourceType " + $DatasourceType
+				throw $errormsg
+			}
+		}
+
+		if ($manifest.allowedSubProtectionPolicyTypes.Count -eq 1) {
+			if (-not $PolicySubType) {
+				if (-not $Policy.PolicyType) {
+					$errormsg = "PolicySubType cannot be empty for AzureVM FullBackup"
```

Comment:
we can avoid using the term fullbackp (for azurevm workload) It is internally full backup but we can avoid calling it out